lib/spack/spack/test/web.py

```diff
@@ -246,14 +247,24 @@ def get_object(self, Bucket=None, Key=None):
 
 
 def test_gather_s3_information(monkeypatch, capfd):
-    mock_connection_data = {
-        "access_token": "AAAAAAA",
-        "profile": "SPacKDeV",
-        "access_pair": ("SPA", "CK"),
-        "endpoint_url": "https://127.0.0.1:8888",
-    }
+    mirror = spack.mirror.Mirror.from_dict(
+        {
+            "fetch": {
```

Comment:
Something I'm confused about is why you changed from an arbitrary dictionary of connection data (`mock_connection_data`) to this thing that is actually a `Mirror`.  For one thing, my understanding was that the `endpoint_url` was optional (and only used/needed for S3 buckets that don't live in AWS S3, but rather `minio` or similar).  If this were a real S3 mirror config, would it have another url?  I.e. the `s3:\\` url?

Also, I'm surprised it's valid to specify all three auth methods simultaneously, I would have thought you'd want to specify only one of `access_token`, `profile`, or `access_pair`.  Maybe specifying all three just helps with testing though?

Reply:
Many different urls correspond to one (or no) mirror, so I thought it was an easier key to use